packages/server/lib/crons/usage.ts

```diff
@@ -0,0 +1,88 @@
+import tracer from 'dd-trace';
+import * as cron from 'node-cron';
+
+import { records } from '@nangohq/records';
+import { connectionService } from '@nangohq/shared';
+import { getLogger, metrics, report, flagHasUsage } from '@nangohq/utils';
+
+import { envs } from '../env.js';
+
+const logger = getLogger('cron.exportUsage');
+const cronMinutes = envs.CRON_EXPORT_USAGE_MINUTES;
+
+export function exportUsageCron(): void {
+    if (!flagHasUsage || cronMinutes <= 0) {
+        logger.info(`Skipping (flagHasUsage=${flagHasUsage}, cronMinutes=${cronMinutes})`);
+        return;
+    }
+    // add some jitter to avoid all instances running at the same time
+    const jitter = Math.floor(Math.random() * cronMinutes);
+    cron.schedule(`*/${cronMinutes + jitter} * * * *`, () => {
+        (async () => {
+            await exec();
+        })();
+    });
+}
+
+export async function exec(): Promise<void> {
+    await tracer.trace<Promise<void>>('nango.cron.exportUsage', async () => {
+        logger.info(`Starting`);
+        await billing.exportBillableConnections();
+        await observability.exportConnectionsMetrics();
+        await observability.exportRecordsMetrics();
+        logger.info(`✅ done`);
+    });
+}
+
+const observability = {
+    exportConnectionsMetrics: async (): Promise<void> => {
+        await tracer.trace<Promise<void>>('nango.cron.exportUsage.observability.connections', async (span) => {
+            try {
+                const connRes = await connectionService.countMetric();
+                if (connRes.isErr()) {
+                    throw connRes.error;
+                }
```

Comment:
if you always throw you could just unwrap but that's just me :D

Reply:
indeed but I like to follow the principle to never `unwrap` in prod code. (only tests or one-off script)